solidity/contracts/bridge/BitcoinTx.sol

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: MIT
+
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+
+pragma solidity 0.8.4;
+
+/// @title Bitcoin transaction
+/// @notice Allows to reference Bitcoin raw transaction in Solidity.
+/// @dev See https://developer.bitcoin.org/reference/transactions.html#raw-transaction-format
+///
+///      Raw Bitcon transaction data:
+///
+///      | Bytes  |     Name     |        BTC type        |        Description        |
+///      |--------|--------------|------------------------|---------------------------|
+///      | 4      | version      | int32_t (LE)           | TX version number         |
+///      | varies | tx_in_count  | compactSize uint (LE)  | Number of TX inputs       |
```

Comment:
Just a remark regarding all `compactSize uint (LE)` mentions. The entire `compactSize` uint is not always LE (source https://developer.bitcoin.org/reference/transactions.html#compactsize-unsigned-integers). 
- If the `compactSize` uint is only one-byte length, that byte is interpreted as the number
- If the `compactSize` uint is longer than one byte, the first byte is the flag saying how many subsequent bytes represent the number. Only those subsequent bytes are little-endian.

I think it's worth clarifying that.

Reply:
I think we can still say it's LE 🤔 For >= 0 && <= 252 it does not matter because we have only one byte so there is no LE/BE. For anything >=253 we have a flag byte and at least two bytes that are LE. See https://github.com/keep-network/tbtc-v2/pull/107/commits/a620f7781d15d2191e1d6265477d24417dd836d8 please.

Reply:
Yeah, I mean mostly the trick with the flag which should not be taken while converting from bytes to uint. The additional explanation you pushed seems to be enough to make it clear.